src/dra/5D6C4.c

```diff
@@ -592,199 +592,155 @@ void func_800FF708(s32 arg0, s32 arg1) {
     g_Status.equipment[arg1 + 2] = rnd;
 }
 
-// DECOMP_ME_WIP func_800FF7B8 https://decomp.me/scratch/Ti1u1
-#ifndef NON_EQUIVALENT
-INCLUDE_ASM("asm/us/dra/nonmatchings/5D6C4", func_800FF7B8);
-#else
-extern s8 D_8006C373[];
-typedef struct {
-    // part of a larger struct, maybe part of PlayerStats?
-    s32 subWeapon;
-} Unkstruct_80097BFC;
-extern Unkstruct_80097BFC subWeapon; // g_SubWeapon
-extern s32 D_800A872C[];
-
-void func_800FF7B8(s32 arg0) {
-    s32 temp_v0;
-    s32 var_a0_2;
-    s32 var_s0;
-    s32 var_s0_10;
-    s32 var_s0_11;
-    s32 var_s0_12;
-    s32 var_s0_13;
-    s32 var_s0_14;
-    s32 var_s0_15;
+void InitStatsAndGear(bool debugMode) {
+    s32 dracDefeatTime;
+    s32 prologueBonusState;
     s32 i;
-    s32 var_s0_2;
-    s32 var_s0_3;
-    s32 var_s0_6;
-    s32 var_s0_7;
-    s32 var_s0_8;
-    s32 var_s0_9;
-    s32 var_v0_3;
-    s32 var_v1_2;
-    s32* var_a1;
-    s32* var_v0_4;
-    s32* var_v0_10;
-    s32* var_v1;
-    s32* var_v1_4;
-    s8* var_v1_5;
-    s32 var_s0_5;
-    s8* var_a0;
-    s8* var_v0;
-    s8* var_v0_2;
-    s8* var_v0_5;
-    s8* var_v0_6;
-    s8* var_v1_3;
-    u8 var_s0_4;
-    u8* var_a1_2;
-    u8* var_a1_3;
-    u8 temp_var_1;
-    u8 temp_var_2;
-    u8 temp_var_3;
-    s32* thingPtr;
-    int new_var;
+    u8* namePtr;
 
     if (D_8003C730 != 0) {
         func_800F53A4();
         func_800FF60C();
         return;
     }
-    if (arg0 == 1) { // First block fully matching
+    if (debugMode == 1) {
```

Comment:
```suggestion
if (debugMode == true)

Reply:
since it's boolean, or just if (debugMode) if you prefer, just a nit without too much importance